changes/ocd-4373.sql

```diff
@@ -0,0 +1,323 @@
+		where s.regulatory_text_citation = p_regulatory_text;
+
+		raise notice 'Standard Exists: %', v_standard;--p_regulatory_text;
+	end if;
+
+	foreach v_criterion_id in array p_criterion_ids
+	loop
+
+		select *
+		into v_certification_criterion
+		from openchpl.certification_criterion
+		where certification_criterion_id = v_criterion_id;
+
+		select count(*)
+		into v_standard_criterion_map_cnt
+		from openchpl.standard_criteria_map where standard_id = v_standard.id
+		and certification_criterion_id = v_criterion_id;
+
+		if v_standard_criterion_map_cnt = 0 then
+
+			insert into openchpl.standard_criteria_map(standard_id, certification_criterion_id, last_modified_user)
+			select v_standard.id, v_criterion_id, -1
+			where not exists (
+				select *
+				from openchpl.standard_criteria_map where standard_id = v_standard.id
+				and certification_criterion_id = v_criterion_id);
+
+			raise notice 'Inserted Standard Crtieria Map: % -- % (%)', p_regulatory_text, v_certification_criterion.number, v_criterion_id;
+
+		else
+			raise notice 'Standard Crtieria Map Exists: % -- % (%)', p_regulatory_text, v_certification_criterion.number, v_criterion_id;
+		end if;
+
+		for v_cert_result in select cpd.chpl_product_number, cpd.certification_date, cert_result.*
+		from openchpl.certification_result cert_result
+			inner join openchpl.certified_product_details cpd
+				on cert_result.certified_product_id = cpd.certified_product_id
+		where cert_result.certification_criterion_id = v_criterion_id
+		and cpd.certification_date between v_standard.required_day and coalesce(v_standard.end_day, '2099-12-31')
+		and cert_result.deleted = false
+		loop
+
+			select count(*)
+			into v_cert_result_standard_cnt
+			from openchpl.certification_result_standard
+			where certification_result_id = v_cert_result.certification_result_id
+			and standard_id = v_standard.id;
+
+			if v_cert_result_standard_cnt = 0 then
+				insert into openchpl.certification_result_standard(standard_id, certification_result_id, last_modified_user)
+				select v_standard.id, v_cert_result.certification_result_id, -1
+				where not exists (
+					select *
+					from openchpl.certification_result_standard
+					where certification_result_id = v_cert_result.certification_result_id
+					and standard_id = v_standard.id);
+
+				raise notice 'Inserted Certification Result Standard %  -- %  -- %', v_cert_result.chpl_product_number, v_certification_criterion.number, v_standard.regulatory_text_citation;
+			else
+				raise notice 'Certification Result Standard Exists %  -- %  -- %', v_cert_result.chpl_product_number, v_certification_criterion.number, v_standard.regulatory_text_citation;
+			end if;
+		end loop;
+	end loop;
+end $$;
+
+
+call openchpl.backfill_standards('170.202(a)(2)', 'Direct Project: ONC Applicability Statement for Secure Health Transport, Version 1.2 August 2015', 3, '2016-01-14', '2016-01-14', null, null, 'Aplies to both b1 and b1 cures', array[16, 59, 60, 165]);
```

Comment:
```suggestion
call openchpl.backfill_standards('170.202(a)(2)', 'Direct Project: ONC Applicability Statement for Secure Health Transport, Version 1.2 August 2015', 3, '2016-01-14', '2016-01-14', null, null, 'Applies to both b1 and b1 cures', array[16, 59, 60, 165]);
```

Reply:
Also, in the spreadsheet that was attached to the email you recently sent out, the "Additional Information" about applying to b1 and b1Cures doesn't appear to be attached to the h1 and h2 criteria. At least, they are different rows in the spreadsheet. Does the Additional Information matter for our purposes?

Reply:
I removed the text from the Additional Info column - I think it was in the wrong column.